classes/plugins/PubIdPlugin.inc.php

```diff
@@ -32,333 +31,228 @@ function PubIdPlugin() {
 	 * @copydoc Plugin::register()
 	 */
 	function register($category, $path) {
-		if (!parent::register($category, $path)) return false;
-		// Enable storage of additional fields.
-		foreach($this->_getDAOs() as $daoName) {
-			HookRegistry::register(strtolower_codesafe($daoName).'::getAdditionalFieldNames', array($this, 'getAdditionalFieldNames'));
-		}
-		return true;
-	}
-
- 	/**
-	 * @copydoc Plugin::manage()
-	 */
-	function manage($args, $request) {
-		$notificationManager = new NotificationManager();
-		$user = $request->getUser();
-		$journal = $request->getJournal();
-
-		$settingsFormName = $this->getSettingsFormName();
-		$settingsFormNameParts = explode('.', $settingsFormName);
-		$settingsFormClassName = array_pop($settingsFormNameParts);
-		$this->import($settingsFormName);
-		$form = new $settingsFormClassName($this, $journal->getId());
-		if ($request->getUserVar('save')) {
-			$form->readInputData();
-			if ($form->validate()) {
-				$form->execute();
-				$notificationManager->createTrivialNotification($user->getId(), NOTIFICATION_TYPE_SUCCESS);
-				return new JSONMessage(true);
-			} else {
-				return new JSONMessage(true, $form->fetch($request));
-			}
-		} elseif ($request->getUserVar('clearPubIds')) {
-			$journalDao = DAORegistry::getDAO('JournalDAO');
-			$journalDao->deleteAllPubIds($journal->getId(), $this->getPubIdType());
-			$notificationManager->createTrivialNotification($user->getId(), NOTIFICATION_TYPE_SUCCESS);
-			return new JSONMessage(true);
-		} else {
-			$form->initData();
-			return new JSONMessage(true, $form->fetch($request));
-		}
+		return parent::register($category, $path);
 	}
 
-
 	//
-	// Protected template methods to be implemented by sub-classes.
+	// Protected template methods from PKPPlubIdPlugin
 	//
 	/**
-	 * Get the public identifier.
-	 * @param $pubObject object
-	 *  (Issue, Article, PublishedArticle, ArticleGalley)
-	 * @param $preview boolean
-	 *  when true, the public identifier will not be stored
-	 * @return string
+	 * @copydoc PKPPubIdPlugin::getPubObjectTypes()
 	 */
-	abstract function getPubId($pubObject, $preview = false);
+	function getPubObjectTypes() {
+		return array('Issue', 'Article', 'ArticleGalley');
```

Comment:
I made this comment on the OMP repo but -- we'll have another level to consider, as we've split Galleys and Files into two different levels now.

Note that in the pkp-lib repo, we refer to Publication Formats (OMP) and Galleys (OJS) as "Representations".